pkg/minikube/bootstrapper/bsutil/featuregates_test.go

```diff
@@ -30,18 +30,18 @@ func TestParseFeatureArgs(t *testing.T) {
 		expectedComponentFeatureArgs string
 	}{
 		{
-			description:  "CoreDNS enabled",
-			featureGates: "CoreDNS=true",
+			description:  "IPv6DualStack enabled",
+			featureGates: "IPv6DualStack=true",
 			expectedKubeadmFeatureArgs: map[string]bool{
-				"CoreDNS": true,
+				"IPv6DualStack": true,
```

Comment:
I am not convinced these values are correct can you link in comment where the official site says the coredns is renamed to ipv6dualstack ?

Reply:
Hey @medyagh I came across this issue while running the tests. And was referring to release notes v1.18
https://kubernetes.io/docs/setup/release/notes/

```kubeadm: remove the deprecated CoreDNS feature-gate. It was set to "true" since v1.11 when the feature went GA. In v1.13 it was marked as deprecated and hidden from the CLI. (#87400, @neolit123) [SIG Cluster Lifecycle]```

Checked this PR - https://github.com/kubernetes/kubernetes/pull/87400 - https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecation.

I will see if I am missing to understand anything else from my end as well.

Reply:
@medyagh - CoreDNS is not renamed to IPv6DualStack.
In Kubeadm, CoreDNS feature gate is deprecated and removed. This particular test in minikube-bsutil is related to feature gates. So it makes sense to replace with IPv6DualStack.
https://kubernetes.io/docs/setup/release/notes/

Reply:
so that would make a problem if a user wants to deploy an older kubernetes version to minikube ?

Reply:
Hi @medyagh Apologies for delayed responses -  i was away for a while.

It is indeed not backward compatible if,  the user starts cluster with feature gates option ```--feature-gates CoreDNS=true``` with this dependency upgrade as the kubeadm refers to v1.18.5

But I was able to start the cluster successfully with older version of kubernetes with this change.
./out/minikube start --kubernetes-version v1.17.0 ==> this is ok.

Haven't tried something like supporting multiple versions in go mod.Not sure if that is an optimal solution to this as I still exploring the architecture and learning.

Also, I  noticed that i should be bumping to 1.18.3  and not 1.18.5.
DefaultKubernetesVersion = "v1.18.3". - Thsi si seperate.

Reply:
Hey @medyagh as far as i have analysed into this, adding k8s as module means adding the subcomponents under require directive and this specific feature gates is part of kubeadm and there is way to support different versions of it. 
i have also looking into this PR earlier when its bumped up to 1.15
https://github.com/kubernetes/minikube/pull/4719 - which eventually replaced the older feature gates and added CoreDNS part.
Provided this CoreDNS is depreciated, can we go for another Pr with v18.3 which will have the same impact of backward compatibility when the driver is started that eventually calls kubeadm init feature-gates=CoreDNS.
But the user can start the old driver without this option.

Not sure how to move forward with this.